pkg/controller/admissionchecks/provisioning/cache.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"sync"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+type cache struct {
```

Comment:
Yes, but having another cache uses more memory and can lead to synchronization problems. There should be a reason to have the cache beyond wrapping some functions. A cache for scheduling is justified, because we need to keep track of usage and do snapshots. I don't see that need here.
You can add a wrapper around the client to get a similar code separation.

Reply:
From a memory perspective, since we only cache `AdmissionChecks` having the provisioning controller set and `ProvisioningRequestsConfigs`, I don't think it should be an issue. 
From a synchronization point of view, I don't think it's less reliable that the informer cache.

Reply:
I'm still wondering if we should have this cache. 

> From a synchronization point of view, I don't think it's less reliable that the informer cache.

Certainly, currently, this cache is much more reliable since this is the fresh code and reviewed by many reviewers. However, codes are going to degrade little by little.

I think that the informer cache is more reliable than this one since, in general, libraries used in many places can get feedback and bug reports. So the informer cache would be reliable than this cache from a long-term perspective.

Also, I think we shouldn't reinvent the wheel as much as possible. If existing cache systems don't match kueue use cases, we can implement new cache systems like `/pkg/cache`.

Reply:
Well, one could argue that `/pkg/cache` could be removed :) ... however , cache removed.

Reply:
> Well, one could argue that /pkg/cache could be removed :) ... however , cache removed.

Yes, if it is possible, it should remove `/pkg/cache`. However, completely removing the `/pkg/cache` is hard because actually, many components propagate cached information using `/pkg/cache` with each other.

However, I think that we should make more simply and smaller the `/pkg/cache` as much as possible.

Reply:
The existing `pkg/cache` serves the purpose of caching resource calculations. Yes, it can be removed, but with significant penalty to the runtime.

Reply:
Yes, I also think `pkg/cache` has advantages more than disadvantages. I just wanted to say that I want to avoid the no-need cache, such as the below:

https://github.com/kubernetes-sigs/kueue/blob/525098b838eb28ee241cf1aaee5e53f161d30a0c/pkg/cache/cache.go#L246